File: hw/chuqui/u2/hw12PHPMonolith/models/Ride.php
<?php
require_once __DIR__ . '/../config/db.php';

class Ride {
    public static function getAll() {
        global $ridesCollection;
        return $ridesCollection->find()->toArray();
    }

    public static function create($origin, $destination, $price) {
        global $ridesCollection;
        $ridesCollection->insertOne([
            'Origin' => $origin,
            'Destination' => $destination,
            'Price' => (float)$price,
            'Date' => new MongoDB\BSON\UTCDateTime()
        ]);
    }

    public static function delete($id) {
        global $ridesCollection;
        $ridesCollection->deleteOne(['_id' => new MongoDB\BSON\ObjectId($id)]);
    }

    public static function find($id) {
        global $ridesCollection;
        return $ridesCollection->findOne(['_id' => new MongoDB\BSON\ObjectId($id)]);
    }

    public static function update($id, $origin, $destination, $price) {
        global $ridesCollection;
        $ridesCollection->updateOne(
            ['_id' => new MongoDB\BSON\ObjectId($id)],
            ['$set' => [
                'Origin' => $origin, 
                'Destination' => $destination, 
                'Price' => (float)$price
            ]]
        );
    }
}
?>

File: ws/chuqui/u2/ws10ASPInCloudwsJSP/TaxiApp/Models/Ride.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace TaxiApp.Models;

public class Ride
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string? Id { get; set; }

    public string Origin { get; set; } = null!;
    public string Destination { get; set; } = null!;
    public decimal Price { get; set; }
    
    public DateTime Date { get; set; } = DateTime.Now;
}